.github/workflows/ci.yaml

```diff
@@ -0,0 +1,229 @@
+name: GitHub Actions CI
+
+# Cache is used for all Windows and macOS dependencies (size approx. 1230 * 2 + 1850 = 4310 MiB)
+# Cache is not used for Ubuntu builds, because it already has all dependencies except
+# the appropriate libtorrent version, which only takes 3-5 minutes to build from source anyway
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    types: [edited, opened, reopened, synchronize]
+    branches: [ master ]
+
+env:
+  # Qt: 5.15.0
+  # libtorrent: RC_1_2 HEAD, 1.2.9
```

Comment:
>Qt: 5.15.0
 libtorrent: RC_1_2 HEAD, 1.2.9

CI doesn't look useful enough to me if it doesn't care about the minimum supported versions of the libraries and tools. Since the main contributors often use the latest versions, we risk to merge breaking changes accidentally.
@Chocobo1?

Reply:
@glassez 
Additionally, I don't know why we need more CI doing the same thing...

Reply:
@glassez 

> CI doesn't look useful enough to me if it doesn't care about the minimum supported versions of the libraries and tools. Since the main contributors often use the latest versions, we risk to merge breaking changes accidentally.

Those versions in those comments are only relevant for the builds that use vcpkg (Windows and macOS). It's a comment that is meant to be updated every time `VCPKG_COMMIT` is bumped - it's there as a convenience so that one does not have to go search for which versions of Qt/libtorrent are available with this version of vcpkg.

I don't understand this criticism though, because Travis and AppVeyor currently do the same thing - Windows and macOS builds are built with recent versions of everything (now they are a bit old, but only because sledge hasn't been updating the dependency bundles as they usually do). Releases for these operating systems use the latest versions, anyway. I think there was even a somewhat recent discussion where it was mentioned that it is in fact preferable that CI for these systems have recent versions of the libraries so that deprecations are caught early.

Ensuring qBittorrent works fine with older versions of libraries and toolchain is most relevant for Linux, and more specifically, Ubuntu and the like. Precisely for this reason, there's CI jobs for the 2 currently supported LTS releases, all using the default libraries and toolchain except for libtorrent. If you're wondering "why use the most recent libtorrent release on the Ubuntu CI?", the reason is because even Ubuntu 20.04 does not include _any_ 1.2.x version: https://packages.ubuntu.com/search?suite=focal&arch=any&searchon=names&keywords=libtorrent-rasterbar.

The goal for this PR was to come up with a CI solution that at least matched what Travis and AppVeyor are doing right now (builds for Windows and macOs with recent dependencies, builds for stock Ubuntu 18.04 [except libtorrent]), but fix their most glaring issues (fix macOS builds, provide CI for stock Ubuntu 20.04 [except libtorrent], and get/build the most recent dependencies for Windows/macOS from official sources, rather than relying on a personally maintained bundle).

If you want to add more jobs with older library versions for Windows and macOS, that can be done later. Though I don't see why that would be desirable, given the reasons above. This PR is just meant to be a drop-in replacement for the current CI solutions.

@Chocobo1 

> Additionally, I don't know why we need more CI doing the same thing...

In case it wasn't clear: this is meant to replace both Travis and AppVeyor.

Reply:
>because Travis and AppVeyor currently do the same thing

If you take our current disadvantages as a basis, your changes won't do much good. As a result, we will trade bad for worse.

Reply:
@glassez 

What exactly is "worse"? I believe I've presented a convincing case as to how and why this alternative is better that our existing solutions.